### PR TITLE
Example using a blend of hard type metadata and dicts for adding new …

### PR DESCRIPTION
**Pull request by GearsAD**
…estimates

### CODE REVIEW

**Review by dehann — APPROVED** [review]

**Comment by dehann** [comment]
Looks good to me!

**Comment by dehann** [comment]
part of #147 discussion

**Comment by dehann** [comment]
want to make sure @Affie is happy before we merge the twig, and then to master

**Comment by Affie** [comment]
You can go ahead, I'm not an expert when it comes to julia.
The only thing I've found with types represented as dictionaries is its harder to use. 
Also please define the fields somehow (maybe in documentation), so I know for example :max will be there and the name won't change to something else. eg. :maximum

**Comment by dehann** [comment]
unlikely to have a name change from (eg :max to :maximum), but if it does occur we will do so with deprecation cycle and minor version numbers.

**Review by Affie — COMMENTED** [review]
Maybe just hold off on it not to break IIF and first do the branch. and this for v0.5.0

**Comment by dehann** [comment]
TLDR; hold off on merging this until we have spoken again please.

hi @GearsAD , Johan convinced me on using `Dict{Symbol, <: AnAbstractType}` given new and improved type inference with anonymous static parameters. https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/pull/148#issuecomment-541536774

So shall we change that now or later?  I told Johan later, but I thought this was already merged.  Perhaps we can just do it now before merging this.  #148 already has it the way Johan suggested, we can maybe just introduce the name change to `PointParametericEstimate` etc.

File: src/entities/DFGVariable.jl
Comment (dehann) [inline src/entities/DFGVariable.jl]: `estimateDict::Dict{Symbol, <: AbstractPointParametric}`

or whatever the chosen name for `AbstractPointParametric` is in the end

File: src/entities/DFGVariable.jl
Comment (dehann) [inline src/entities/DFGVariable.jl]: `struct PointParametricEstimates <: AbstractPointParametric` as originally suggested by @Affie.

or whatever the chosen name for `AbstractPointParametric` is in the end

**Review by dehann — CHANGES_REQUESTED** [review]

File: src/entities/DFGVariable.jl
Comment (dehann) [reply src/entities/DFGVariable.jl]: from #148 

`<: AbstractVariableEstimate`

File: src/entities/DFGVariable.jl
Comment (dehann) [reply src/entities/DFGVariable.jl]: from #148

`<: AbstractVariableEstimate`

**Comment by Affie** [comment]
Closed by design decision from #147 [last few comments](https://github.com/JuliaRobotics/DistributedFactorGraphs.jl/issues/147#issuecomment-542091523)

**Pull request closed without merging**
